edgelet/edgelet-kube/src/runtime.rs

```diff
@@ -236,11 +237,50 @@ where
     }
 
     fn system_info(&self) -> Self::SystemInfoFuture {
-        // TODO: Implement this.
-        Box::new(future::ok(SystemInfo::new(
-            "linux".to_string(),
-            "x86_64".to_string(),
-        )))
+        #[derive(Debug, serde_derive::Serialize)]
+        pub struct Architecture {
+            name: String,
+            nodes_count: u32,
+        };
+
+        let fut = self
+            .client
+            .lock()
+            .expect("Unexpected lock error")
+            .borrow_mut()
+            .list_nodes()
+            .map_err(|err| Error::from(err.context(ErrorKind::SystemInfo)))
```

Comment:
```suggestion
            .map_err(|err| Error::from(err.context(ErrorKind::RuntimeOperation(RuntimeOperation::SystemInfo)))
```
And no need in `ErrorKind::SystemInfo` anymore